crates/polars-core/src/chunked_array/logical/categorical/merge.rs

```diff
@@ -192,6 +199,33 @@ pub fn call_categorical_merge_operation<I: CategoricalMergeOperation>(
     }
 }
 
+struct DoNothing;
+impl CategoricalMergeOperation for DoNothing {
+    fn finish(self, _lhs: &UInt32Chunked, rhs: &UInt32Chunked) -> PolarsResult<UInt32Chunked> {
+        Ok(rhs.clone())
+    }
+}
+
+// Make the right categorical compatible with the left
+pub fn make_categoricals_compatible(
+    ca_left: &CategoricalChunked,
+    ca_right: &CategoricalChunked,
+) -> PolarsResult<(CategoricalChunked, CategoricalChunked)> {
```

Comment:
Not for this PR, but can you make `call_categorical_merge_operation` produce less code bloat by changing to dynamic dispatch for the finish:

```rust
pub fn call_categorical_merge_operation<I: CategoricalMergeOperation>(
    cat_left: &CategoricalChunked,
    cat_right: &CategoricalChunked,
    merge_ops: I,
) -> PolarsResult<CategoricalChunked> 


TO:

pub fn call_categorical_merge_operation(
    cat_left: &CategoricalChunked,
    cat_right: &CategoricalChunked,
    merge_ops: &dyn CategoricalMergeOperation,
) -> PolarsResult<CategoricalChunked> {
```